### PR TITLE
Phase two

### PR DESCRIPTION
**Pull request by apope1**
We finished the second phase.

### CODE REVIEW

File: App/App/UI/PhaseTwo/AuthorMain.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                MessageBox.Show("Please select a file to upload");
+            else
+            {  
+                
+                
+                MetaInformation mt = new MetaInformation(proposal, controller, true, textBox1.Text);
+                mt.Show();
+
+                
+            }
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/AuthorMain.cs]: Remove white lines

File: App/App/UI/PhaseTwo/AuthorMain.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            //refreshProposals();
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/AuthorMain.cs]: Why is this commented?

File: App/App/UI/PhaseTwo/ChairAssignementToReviewer.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            {
+                User reviewer = phaseTwoController.getReviewerById(Int32.Parse(row.Cells[0].ToString()));
+                reviewers.Add(reviewer);
+            }
+            foreach(Review rev in reviews)
+            {
+                if(!reviewers.Contains(rev.Reviewer))
+                {
+                    phaseTwoController.removeReview(rev.ReviewId);
+                    continue;
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/ChairAssignementToReviewer.cs]: This "continue" has no effect. You can remove it.

File: App/App/UI/PhaseTwo/ChairMain.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        public ChairMain(PhaseTwoController phaseTwoController)
+        {
+            this.phaseTwoController = phaseTwoController;
+            InitializeComponent();
+        }
+
+        private void sendEmailsButton_Click(object sender, EventArgs e)
+        {
+            ChairToAcceptedAuthors sendMailView = new ChairToAcceptedAuthors(phaseTwoController, phaseTwoController.getMailSender);
+            sendMailView.Show();
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/ChairMain.cs]: Hide the current form and show the child form in the same location as the parent.

File: App/App/UI/PhaseTwo/ChairMain.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace App.UI.PhaseTwo
+{
+    public partial class ChairMain : Form
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/ChairMain.cs]: Docs

File: App/App/UI/PhaseTwo/ChairMain.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        private void sendEmailsButton_Click(object sender, EventArgs e)
+        {
+            ChairToAcceptedAuthors sendMailView = new ChairToAcceptedAuthors(phaseTwoController, phaseTwoController.getMailSender);
+            sendMailView.Show();
+        }
+
+        private void assignButton_Click(object sender, EventArgs e)
+        {
+            ChairAssignementToReviewer assignmentView = new ChairAssignementToReviewer(phaseTwoController);
+            assignmentView.Show();
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/ChairMain.cs]: Hide the current form and show the child form in the same location as the parent.

File: App/App/UI/PhaseTwo/ChairToAcceptedAuthors.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+using System.Net.Mail;
+using System.Windows.Forms;
+
+/// <summary>
+/// Author : Diana Gociu
+/// </summary>
+
+namespace App.UI.PhaseTwo
+{
+    public partial class ChairToAcceptedAuthors : Form
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/ChairToAcceptedAuthors.cs]: Docs. Above the class, not the namespace

File: App/App/UI/PhaseTwo/PCMemberBidProposals.cs [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+using System.Linq;
+using System.Windows.Forms;
+
+/// <summary>
+/// Author : Vancea Vladut
+/// </summary>
+
+namespace App.UI.PhaseTwo
+{
+    public partial class PCMemberBidProposals : Form
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberBidProposals.cs]: Docs are for the class, not the whole namespace

File: App/App/UI/PhaseTwo/PCMemberBidProposals.cs [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void submitButton_Click(object sender, EventArgs e)
+        {
+            RadioButton checkedReviewButton = this.Controls.OfType<RadioButton>().FirstOrDefault(r => r.Checked);
+            Proposal prop = phaseTwoController.findProposalById(int.Parse(proposalsDataGridView.CurrentRow.Cells[0].Value.ToString()));
+
+            if (checkedReviewButton.Text.Equals("I want") || checkedReviewButton.Text.Equals("I can"))
+            {
+                //Real author: Andu Popa...
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberBidProposals.cs]: I appreciate it, but add it to the class doc :)

File: App/App/UI/PhaseTwo/PCMemberBidProposals.cs [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                Review review = new Review("", "", reviewer, prop, new DateTime(2000,01,01));
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberBidProposals.cs]: 2000, 01, 01?

File: App/App/UI/PhaseTwo/PCMemberBidProposals.cs [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                phaseTwoController.addReview(review);
+            }
+            else
+            {
+                //phaseTwoController.deleteProposalFromDB(prop);
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberBidProposals.cs]: Why is this commented?

File: App/App/UI/PhaseTwo/PCMemberBidProposals.cs [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+            }
+
+
+            
+
+        }
+
+        private void proposalsDataGridView_CellContentClick(object sender, DataGridViewCellEventArgs e)
+        {
+            //do nothing
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberBidProposals.cs]: Remove it if it doesn't do anything :D

File: App/App/UI/PhaseTwo/PCMemberMain.cs [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace App.UI.PhaseTwo
+{
+    public partial class PCMemberMain : Form
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberMain.cs]: Docs

File: App/App/UI/PhaseTwo/PCMemberMain.cs [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+        public PCMemberMain(PhaseTwoController phaseTwoController, User reviewer)
+        {
+            this.phaseTwoController = phaseTwoController;
+            this.reviewer = reviewer;
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// Author: Andu Popa
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberMain.cs]: Docs go to the class, not to the methods

File: App/App/UI/PhaseTwo/PCMemberMain.cs [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void rateButton_Click(object sender, EventArgs e)
+        {
+            PCMemberRateProposals rateView = new PCMemberRateProposals(phaseTwoController.findProposalByIdUser(reviewer.UserId), reviewer);
+            rateView.Show();
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberMain.cs]: Hide the parent form and show the child in the same place

File: App/App/UI/PhaseTwo/PCMemberMain.cs [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+        /// <param name="e"></param>
+        private void buttonRefreshReviews_Click(object sender, EventArgs e)
+        {
+            if(phaseTwoController.findReviewByIdUser(reviewer.UserId).Count > 0)
+            {
+                initRatingsGridView();
+            }
+        }
+        /// <summary>
+        /// Author: Andu Popa
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberMain.cs]: Docs go to the class, not to the methods

File: App/App/UI/PhaseTwo/PCMemberMain.cs [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            BindingList<Review> bindingList = new BindingList<Review>(phaseTwoController.findReviewByIdReviewer(reviewer.UserId));
+            BindingSource source = new BindingSource(bindingList, null);
+            ratingsGridView.DataSource = source;
+        }
+
+        private void bidButton_Click(object sender, EventArgs e)
+        {
+            PCMemberBidProposals bidView = new PCMemberBidProposals(phaseTwoController, reviewer);
+            bidView.Show();
```
Comment (ovidiuenache) [inline App/App/UI/PhaseTwo/PCMemberMain.cs]: Hide the parent class and show the child in the same place as the parent

File: App/App/UI/PhaseTwo/PCMemberBidProposals.cs (hunk 10 above)
Comment (apope1) [inline App/App/UI/PhaseTwo/PCMemberBidProposals.cs]: I took this for granted as an invalid date (like null or something like that).

**Pull request merged**